File: app/src/main/java/org/record/tiny/ui/view/MainActivity.java
package org.record.tiny.ui.view;

import android.os.Bundle;
import android.widget.Button;

import com.apkfuns.logutils.LogUtils;

import org.record.tiny.R;
import org.record.tiny.base.BaseActivity;
import org.record.tiny.utils.RxIntent;

import java.util.HashMap;

import butterknife.BindView;
import io.reactivex.functions.Consumer;

@SuppressWarnings("All")
public class MainActivity extends BaseActivity {
    @BindView(R.id.but)
    Button mbutton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        //addFragment(R.id.activity_main_layout, DisplayFragment.newInstance());

        RxIntent.getInstance().subscribe(new Consumer<HashMap>() {
            @Override
            public void accept(HashMap hashMap) throws Exception {
                LogUtils.d("MainActivity -> accept: " + hashMap);
            }
        });
    }
}

File: app/src/main/java/org/tiny/record/MainActivity.kt
package org.tiny.record

import android.databinding.DataBindingUtil
import android.os.Bundle
import com.apkfuns.logutils.LogUtils
import com.trello.rxlifecycle2.android.ActivityEvent
import com.trello.rxlifecycle2.components.RxActivity
import io.reactivex.Flowable
import io.reactivex.android.schedulers.AndroidSchedulers
import org.tiny.record.databinding.ActivityMainBinding
import java.util.concurrent.TimeUnit

class MainActivity : RxActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding: ActivityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        Flowable.interval(3, TimeUnit.SECONDS)
                .compose(bindUntilEvent(ActivityEvent.DESTROY))
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { timer ->
                        binding.model = TestViewModel("" + System.currentTimeMillis())
                        LogUtils.d("haha" + timer)
                }

    }
}
